File: RedSheet.DbEntities/Models/ProjectCulturePlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectCulturePlans", Schema = "dbo")]
	[Serializable]
	public partial class ProjectCulturePlan  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectCulturePlanId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ProjectCulturePlanValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Nullable<CulturePlanCategory> CulturePlanCategoryId { get; set; } 
 
	[RelationshipTableAttribue("CulturePlans","dbo")]
	//Relationships
	public Nullable<int> CulturePlanId { get; set; } 
 
	[ForeignKey("CulturePlanId")]
	public CulturePlan CulturePlan { get; set; } 

	public ProjectCulturePlan()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/project-culture-plan.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { CulturePlan,  vProjectCulturePlanRecord  } from './'
export class ProjectCulturePlan {
    constructor(projectCulturePlan?: ProjectCulturePlan  | vProjectCulturePlanRecord )  {
        let properties = [ "projectCulturePlanId", "projectCulturePlanValue", "projectModuleId", "culturePlanCategoryId", "culturePlanId",];
        for (let property of properties)
            if (projectCulturePlan && projectCulturePlan[property])
                this[property] = projectCulturePlan[property];
    }
 
	projectCulturePlanId : number =   0 ;
 
    @required()
    @maxLength(200)
	projectCulturePlanValue : string =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
	culturePlanCategoryId : number =   undefined;
 
	culturePlanId : number =   undefined;
	culturePlan : CulturePlan  ;


}
